README.md

```diff
@@ -151,149 +136,73 @@ Under this naming convention, `DataSet` simply refers to a graph of `DataPipes`,
+shuffled/batched more than once. You should choose one or the other.
 
-```py
-class MapperIterDataPipe(IterDataPipe):
-    ...
-
-    def __iter__(self):
-        for d in self.dp:
-            yield self.fn(d)
-```
-
-#### Length
-
-In many cases, as in our `MapperIterDataPipe` example, the `__len__` method of a DataPipe returns the length of the
-source DataPipe.
-
-```py
-class MapperIterDataPipe(IterDataPipe):
-    ...
-
-    def __len__(self):
-        return len(self.dp)
-```
+Q: How is multiprocessing handled with DataPipes?
 
-However, note that `__len__` is optional for `IterDataPipe` and often inadvisable. For `CSVParserIterDataPipe` in the
-[using DataPipes section below](#using-datapipes), `__len__` is not implemented because the number of rows in each file
-is unknown before loading it. In some special cases, `__len__` can be made to either return an integer or raise an Error
-depending on the input. In those cases, the Error must be a `TypeError` to support Python's build-in functions like
-`list(dp)`.
+A: Multi-process data loading is still handed by DataLoader, see the
+[documentation here for details](https://pytorch.org/docs/stable/data.html#single-and-multi-process-data-loading).
 
-#### Registering DataPipes with the functional API
+Q: What is the upcoming plan for DataLoader?
 
-Each DataPipe can be registered to support functional invocation using the decorator `functional_datapipe`.
+A: There will be a new version of DataLoader in the next release. At the high level, the plan is that DataLoader V2 will
+only be responsible for multiprocessing, distributed, and similar functionalities, not data processing logic. All data
+processing features, such as the shuffling and batching, will be moved out of DataLoader to DataPipe. At the same time,
+the current/old version of DataLoader should still be available and you can use DataPipes with that as well.
 
-```py
-@functional_datapipe("map")
-class MapperIterDataPipe(IterDataPipe):
-    ...
-```
-
-The stack of DataPipes can then be constructed in functional form:
-
-```py
->>> import torchdata.datapipes as dp
->>> datapipes1 = dp.iter.FileOpener(['a.file', 'b.file']).map(fn=decoder).shuffle().batch(2)
-
->>> datapipes2 = dp.iter.FileOpener(['a.file', 'b.file'])
->>> datapipes2 = dp.iter.Mapper(datapipes2)
->>> datapipes2 = dp.iter.Shuffler(datapipes2)
->>> datapipes2 = dp.iter.Batcher(datapipes2, 2)
-```
-
-In the above example, `datapipes1` and `datapipes2` represent the exact same stack of `IterDataPipe`s.
-
-### Using DataPipes
-
-For a complete example, suppose we want to load data from CSV files with the following steps:
+## Contributing
 
-- List all csv files in a directory
-- Load csv files
-- Parse csv file and yield rows
+We welcome PRs! See the [CONTRIBUTING](CONTRIBUTING.md) file.
 
-To support the above pipeline, `CSVParser` is registered as `parse_csv` to consume file streams and expand them as rows.
+## Building the Documentation
```

Comment:
Can you move this part into https://github.com/pytorch/data/blob/main/docs/README.md?

Reply:
Done